runtime/common/CMakeLists.txt

```diff
@@ -77,6 +77,63 @@ if(EXISTS "$ENV{CURL_INSTALL_PREFIX}/cacert.pem")
   install(FILES "$ENV{CURL_INSTALL_PREFIX}/cacert.pem" DESTINATION .)
 endif()
 
+## ----- MLIR-dependent core components -----
+set(LIBRARY_NAME cudaq-mlir-core)
+add_library(${LIBRARY_NAME} SHARED
+  RuntimeMLIR.cpp
+  JIT.cpp
+)
+target_include_directories(${LIBRARY_NAME}
+  PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/runtime>
+    $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+)
+target_link_libraries(${LIBRARY_NAME}
+  PRIVATE
+    cudaq-common
+    CCDialect
+    QuakeDialect    
+    OptTransforms
+    OptCodeGen
+    MLIRParser
+    MLIRIR
+    MLIRExecutionEngine
+    MLIRTranslateLib
+    MLIRSupport 
+    MLIROptLib
+    MLIRTransforms
+    MLIRTargetLLVMIRExport
+    MLIRLLVMCommonConversion
+    MLIRLLVMToLLVMIRTranslation
+    fmt::fmt-header-only
+)
+install(TARGETS ${LIBRARY_NAME} EXPORT cudaq-targets DESTINATION lib)
+
+## ----- MLIR-dependent all compiler components including argument converter -----
+set(LIBRARY_NAME cudaq-mlir-compiler)
```

Comment:
all this library provides is argument synthesis via the ArgumentConversion? Maybe you should call it cudaq-arg-synth or something that isn't as general as mlir-compiler?